### PR TITLE
Consistent spec titles

### LINKED ISSUE #608: Spec titles are not consistent in the rendered docs
Spec titles in the TOC in the left-side bar are not consistent. Most are module names and some are descriptive. There are a few similar inconsistencies in the list of modules in the body of the page.

The names in the bar are controlled by the `title` attribute in each spec md document.
specs_toc

### PR DESCRIPTION
**Pull request by milancurcic**
This PR:

* Renames several spec title attributes to make them consistent with others (module names)
* A few similar edits to the index.md of the Spec top-level page, for consistency.

Closes #608.

### CODE REVIEW

**Review by awvwgk — APPROVED** [review]

File: doc/specs/stdlib_hash_procedures.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Hash procedures
+title: hash
```
Comment (jvdp1) [inline doc/specs/stdlib_hash_procedures.md:2]: Actually I would prefer reading `Hash procedures` over `hash`, which is more informative to me.
E..g., `Numerical integration` can be easier understood than `quadrature`.

**Review by jvdp1 — APPROVED** [review]
I approve this PR because it makes all titles using the same rule. However, I would prefer more explicit titles.

File: doc/specs/stdlib_hash_procedures.md (hunk 1 above)
Comment (milancurcic) [reply doc/specs/stdlib_hash_procedures.md:2]: Why don't we choose whether to list descriptive names or module names and use it consistently? I assumed we were listing module names. Can we merge this as is, and I'll open a new issue to discuss whether to switch this table to descriptive names?

Comment (jvdp1) [reply doc/specs/stdlib_hash_procedures.md:2]: Yes, merge it as is. At least it is consistent now. Thank you.

**Pull request merged**